internal/contour/k8s.go

```diff
@@ -89,7 +95,11 @@ func (reh *ResourceEventHandler) OnUpdate(oldObj, newObj interface{}) {
 		reh.WithField("op", "update").Debugf("%T", newObj)
 		reh.Remove(oldObj)
 		reh.Insert(newObj)
-		reh.update()
+		// If secret updated, but is not used in an Ingress or IngressRoute resource then skip
+		// notifying that the cache has changed, but still allow it to be stored for later use
+		if reh.validSecret(newObj) {
```

Comment:
I think this logic needs to be propogated to OnDelete, deleting an unrelated secret shouldn't cause an update.

Reply:
Ahh ok, my initial thought was we wanted to always rebuild, but you're right, an un-referenced secret still shouldn't cause an update.